client/cli/src/lib.rs

```diff
@@ -943,13 +955,22 @@ pub fn fill_import_params<C, G, E>(
 	config.wasm_method = cli.wasm_method.into();
 
 	let exec = &cli.execution_strategies;
-	let exec_all_or = |strat: ExecutionStrategy| exec.execution.unwrap_or(strat).into();
+	let exec_all_or = |strat: ExecutionStrategy, default: ExecutionStrategy| {
+		if strat == default && is_dev {
```

Comment:
We don't really differentiate between flag being provided at the CLI level or not, we would need to switch to `Options`, but it complicates help and default values, so it's most likely not worthy at all.

It seems to do what it says on the tin now.

We could maybe simplify the code a bit to:
```rust
exec.execution.unwrap_or(if is_dev && strat == default {
  ExecutionStrategy::Native
} else {
  strat
})
```